core/src/message/context.rs

```diff
@@ -484,9 +498,8 @@ impl MessageContext {
 
     /// Wake message by it's message id.
     pub fn wake(&mut self, waker_id: MessageId, delay: u32) -> Result<(), Error> {
-        if self.store.awaken.insert(waker_id) {
+        if !self.outcome.awakening.iter().any(|v| v.0 == waker_id) {
```

Comment:
This is a Vec<(MessageId, u64)>, contains does not take a function, but a value to compare with Eq, but we do not care what the u64 is since only if the MessageId is the same, it is a duplicate, therefore, I cannot use contains.